packages/@react-spectrum/dialog/stories/Dialog.stories.tsx

```diff
@@ -350,31 +350,50 @@ function renderWithForm({width = 'auto', ...props}) {
+    <p>Ut porttitor leo a diam sollicitudin tempor id eu nisl. Tristique senectus et netus et malesuada fames ac turpis egestas. Tellus in hac habitasse platea dictumst vestibulum rhoncus est. Integer feugiat scelerisque varius morbi enim nunc faucibus a. Tempus quam pellentesque nec nam aliquam sem et. Quam viverra orci sagittis eu volutpat odio facilisis mauris. Nunc lobortis mattis aliquam faucibus purus in massa tempor. Tincidunt dui ut ornare lectus sit amet est. Magna fermentum iaculis eu non. Posuere sollicitudin aliquam ultrices sagittis orci a scelerisque. Vitae aliquet nec ullamcorper sit amet risus nullam eget felis. Vitae proin sagittis nisl rhoncus mattis rhoncus. Nunc vel risus commodo viverra maecenas. Diam in arcu cursus euismod. Dolor morbi non arcu risus quis varius quam. Amet nisl suscipit adipiscing bibendum. Nulla pellentesque dignissim enim sit amet venenatis. Nunc congue nisi vitae suscipit tellus mauris a diam maecenas. In hac habitasse platea dictumst vestibulum rhoncus est pellentesque elit</p>
+    <p>Cras semper auctor neque vitae tempus quam pellentesque nec. Maecenas ultricies mi eget mauris pharetra et ultrices neque ornare. Vulputate enim nulla aliquet porttitor lacus luctus accumsan tortor posuere. Pellentesque habitant morbi tristique senectus et. Ipsum dolor sit amet consectetur adipiscing elit pellentesque. Sem et tortor consequat id porta nibh venenatis. Viverra nibh cras pulvinar mattis nunc sed blandit. Urna porttitor rhoncus dolor purus. Vivamus arcu felis bibendum ut. Cras sed felis eget velit aliquet. Sed tempus urna et pharetra pharetra. Viverra adipiscing at in tellus integer feugiat scelerisque varius morbi. Ac tortor vitae purus faucibus ornare suspendisse sed nisi lacus. Ultrices neque ornare aenean euismod elementum nisi quis eleifend quam. Vel turpis nunc eget lorem. Quisque egestas diam in arcu cursus euismod quis viverra. At tempor commodo ullamcorper a lacus vestibulum sed. Id aliquet lectus proin nibh nisl condimentum id venenatis. Quis viverra nibh cras pulvinar. Purus in mollis nunc sed.</p>
   </React.Fragment>
 );
 
 function renderLongContent({width = 'auto', ...props}) {
   return (
-    <div style={{display: 'flex', width, margin: '100px 0'}}>
-      <DialogTrigger defaultOpen>
+    <div style={{display: 'flex', flexDirection: 'column', alignItems: 'start', width, padding: '0 20px'}}>
+      <p>
+        Test instructions on mobile:
+        <ol>
+          <li>Scroll down and open dialog.</li>
+          <li>Tap on the top input. Dialog should resize when the keyboard comes up and remain at the top of the screen. The rest of the page should not scroll (the first time you do this it may scroll slightly due to the bottom toolbar showing).</li>
+          <li>Close the software keyboard.</li>
+          <li>Scroll down and tap on the bottom input. Dialog should resize as before, and focused input should scroll into view.</li>
```

Comment:
I noticed in iOS that if you close the virtual keyboard after this step, the bottom input is scrolled out of sight and the scroll position is moved to somewhere in the middle of the top paragraph. Android retains the scroll position so I assume this has something to do with the workarounds for mobile safari in this pull. Is it possible to keep the original scroll position to keep the bottom input in view? Will try and digest the usePreventScroll code some more to see if I can think up of something.

Reply:
Also perhaps as a consequence of the above, try the following:
1. Open the Dialog
2. Click on the bottom input and close the virtual keyboard. Note the bottom input is no longer in view
3. Scroll to the top input and tap on it. Note that the bottom input is now scrolled into view

Reply:
Ah yeah we could possibly try to restore the scroll position when the keyboard closes as well...